File: ulyp-ui/src/main/java/com/ulyp/ui/util/file/TmpFile.java
package com.ulyp.ui.util.file;

import java.io.File;
import java.nio.file.Path;
import java.nio.file.Paths;

public class TmpFile {

    private final Path path;

    public TmpFile(String name) {
        this.path = Paths.get(System.getProperty("java.io.tmpdir"), name);

        new File(name).deleteOnExit();
    }

    public Path getPath() {
        return path;
    }
}

File: ulyp-ui/src/main/kotlin/com/ulyp/ui/util/file/TmpFile.kt
package com.ulyp.ui.util.file

import java.io.File
import java.nio.file.Path
import java.nio.file.Paths

class TmpFile(name: String) {
    val path: Path = Paths.get(System.getProperty("java.io.tmpdir"), name)

    init {
        File(name).deleteOnExit()
    }
}